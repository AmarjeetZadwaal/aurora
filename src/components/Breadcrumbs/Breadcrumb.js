import React from "react";
import PropTypes from "prop-types";

import StyledBreadcrumb from "./Breadcrumb.styles";

const Breadcrumb = ({ children, ...props }) => (
  <StyledBreadcrumb
    itemScope
    itemType="http://schema.org/Breadcrumb"
    {...props}
  >
    {children}
  </StyledBreadcrumb>
);

Breadcrumb.propTypes = {
  children: PropTypes.node.isRequired
};

export default Breadcrumb;
