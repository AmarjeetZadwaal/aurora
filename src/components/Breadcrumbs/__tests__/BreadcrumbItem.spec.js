import React from "react";
import renderer from "react-test-renderer";

import Breadcrumb from "../";

describe("<Breadcrumb.Item />", () => {
  const PROPS = {
    position: "1",
    href: "/",
    children: "Home"
  };

  it("renders correctly", () => {
    const component = renderer.create(<Breadcrumb.Item {...PROPS} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
