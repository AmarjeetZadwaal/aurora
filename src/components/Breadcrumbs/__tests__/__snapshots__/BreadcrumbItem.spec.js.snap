// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb.Item /> renders correctly 1`] = `
.c0 {
  padding: 0 4px;
}

.c0:first-of-type {
  padding-left: 0;
}

.c0:last-of-type {
  padding-right: 0;
}

.c1 {
  font-size: 14px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  color: rgba(2,108,223,1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<li
  className="c0"
  itemProp="itemListElement"
  itemScope={true}
  itemType="http://schema.org/ListItem"
>
  <a
    className="c1"
    color=""
    href="/"
    itemProp="item"
    rel={undefined}
    target=""
  >
    Home
  </a>
  <meta
    content="Home"
    itemProp="name"
  />
  <meta
    content="1"
    itemProp="position"
  />
</li>
`;
